Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py

```diff
@@ -0,0 +1,152 @@
+from typing import List, Optional, Pattern, Match, Dict
+import regex
+from recognizers_text.matcher.match_result import MatchResult
+
+
+class BaseTimeZoneParser(DateTimeParser):
+    @property
+    def parser_name(self) -> str:
+        return Constants.SYS_DATETIME_TIMEZONE
+
+    def compute_minutes(self, utc_offset: str) -> int:
+        pass
+
+    def convert_offset_in_mins_to_offset_string(self, offset_mins: int) -> str:
+        pass
+
+    def convert_mins_to_regular_format(self, offset_mins: int) -> str:
+        pass
+
+    def normalize_text(self, text: str) -> str:
+        pass
+
+    def parse(self, extract_result: ExtractResult, ref_date: datetime = None) -> ParseResult:
+        pass
+
+    def filter_results(self, query: str, candidate_results: List[DateTimeParseResult]) -> List[DateTimeParseResult]:
+        pass
+
+    def parse(self, extract_result: ExtractResult, ref_date: datetime) -> DateTimeParseResult:
+        pass
+
+    def get_datetime_resolution_result(self, offset_mins: int, text: str) -> DateTimeResolutionResult:
+        pass
+
+
+class TimeZoneExtractorConfiguration(DateTimeOptionsConfiguration):
+    @property
+    @abstractmethod
+    def direct_utc_regex(self) -> regex:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def location_time_suffix_regex(self) -> regex:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def location_matcher(self) -> StringMatcher:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def time_zone_matcher(self) -> StringMatcher:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def ambiguous_time_zone_list(self) -> List[str]:
+        raise NotImplementedError
+
+
+class BaseTimeZoneExtractor(DateTimeZoneExtractor):
+    @property
+    def extractor_type_name(self) -> str:
+        return Constants.SYS_DATETIME_TIME
+
+    def __init__(self, config: TimeZoneExtractorConfiguration):
+        self.config = config
+
+    def extract(self, source: str, reference: datetime = None) -> List[ExtractResult]:
+        from .utilities import merge_all_tokens
+        tokens: List[Token] = list()
+        normalized_text = QueryProcessor.remove_diacritics(source)
+        tokens.append(self.match_timezones(normalized_text))
+        tokens.append(self.match_location_times(normalized_text, tokens))
+        return merge_all_tokens(tokens, source, self.extractor_type_name)
+
+    def remove_ambiguous_time_zone(self, extract_result: List[ExtractResult]) -> List[ExtractResult]:
+        return [item for item in extract_result if self.config.ambiguous_time_zone_list in item.text]
+
+    def match_location_times(self, text: str, tokens: List[Token]) -> List[Token]:
+        result: List[Token] = list()
+
+        if not self.config.location_time_suffix_regex:
+            return result
+
+        time_match = regex.match(self.config.location_time_suffix_regex, text)
```

Comment:
re.match() only return a match object, if found. Based on C# version, I think here you have to obtain multiple Match objects.

Reply:
You are right! the original implementation uses `.Matches`, that in C# returns `MatchCollection`. 
Instead, I think we can use this: 
`time_match = list(regex.finditer(self.config.location_time_suffix_regex, text))`
What do you think?